### PR TITLE
docs: docstrings for flow module

### PR DESCRIPTION
**Pull request by cristianmtr**

### CODE REVIEW

File: jina/flow/base.py [hunk 1]
```diff
@@ -64,6 +61,9 @@ def __init__(self, args: Optional['argparse.Namespace'] = None, env: Optional[Di
         will generate 4 Peas, but it will force the :class:`GatewayPea` to take BIND role,
         as the head and tail routers are removed.
 
+        :param kwargs: other keyword arguments that will be shared by all pods in this flow
```
Comment (alexcg1) [inline jina/flow/base.py:64]: - **P**ods, **F**low
- Should we capitalize first letter (i.e. **O**ther)?

File: jina/flow/base.py [hunk 2]
```diff
@@ -171,9 +173,11 @@ def needs(self, needs: Union[Tuple[str], List[str]],
 
     def needs_all(self, name: str = 'joiner', *args, **kwargs) -> 'BaseFlow':
         """
-        Collect all hanging Pod so far and add a blocker to the flow, wait until all handing peas completed.
+        Collect all hanging Pods so far and add a blocker to the Flow; wait until all handing peas completed.
```
Comment (alexcg1) [inline jina/flow/base.py:176]: Peas

**Review by alexcg1 — CHANGES_REQUESTED** [review]
Looks fine generally. Just check capitalization of Pods, Flows, Peas throughout (I've commented initially but don't have time to go through each one - there are a few tho)

File: jina/flow/base.py (hunk 1 above)
Comment (cristianmtr) [reply jina/flow/base.py]: Transformed to uppercase most of them (in this folder)

Comment (cristianmtr) [reply jina/flow/base.py]: > Should we capitalize first letter (i.e. Other)?

I don't think we're doing this atm. Also the linter doesn't complain about it, so I wouldn't care about it too much

File: jina/flow/base.py [hunk 3]
```diff
@@ -256,8 +270,8 @@ def add(self,
     def inspect(self, name: str = 'inspect', *args, **kwargs) -> 'BaseFlow':
         """Add an inspection on the last changed Pod in the Flow
 
-        Internally, it adds two pods to the flow. But no worry, the overhead is minimized and you
-        can remove them by simply give `Flow(inspect=FlowInspectType.REMOVE)` before using the flow.
+        Internally, it adds two Pods to the Flow. But no worry, the overhead is minimized and you
```
Comment (alexcg1) [inline jina/flow/base.py:273]: `no worry` -> `don't worry`

File: jina/flow/base.py [hunk 4]
```diff
@@ -256,8 +270,8 @@ def add(self,
...
+        can remove them by simply give `Flow(inspect=FlowInspectType.REMOVE)` before using the Flow.
```
Comment (alexcg1) [inline jina/flow/base.py:274]: `give` -> `using`

File: jina/flow/base.py [hunk 5]
```diff
@@ -267,17 +281,20 @@ def inspect(self, name: str = 'inspect', *args, **kwargs) -> 'BaseFlow':
                     -- PUB-SUB -- InspectPod (Hanging)
 
         In this way, :class:`InspectPod` looks like a simple ``_pass`` from outside and
-        does not introduce side-effect (e.g. changing the socket type) to the original flow.
+        does not introduce side-effect (e.g. changing the socket type) to the original Flow.
```
Comment (alexcg1) [inline jina/flow/base.py:284]: side-effect**s**

File: jina/flow/base.py [hunk 6]
```diff
@@ -267,17 +281,20 @@ def inspect(self, name: str = 'inspect', *args, **kwargs) -> 'BaseFlow':
...
         The original incoming and outgoing socket types are preserved.
 
-        This function is very handy for introducing evaluator into the flow.
+        This function is very handy for introducing evaluator into the Flow.
```
Comment (alexcg1) [inline jina/flow/base.py:287]: an Evaluator

File: jina/flow/base.py [hunk 7]
```diff
@@ -449,20 +476,23 @@ def start(self):
...
     def num_pods(self) -> int:
-        """Get the number of pods in this flow"""
+        """Get the number of Pods in this Flow
+        # noqa: DAR201"""
         return len(self._pod_nodes)
 
     @property
     def num_peas(self) -> int:
-        """Get the number of peas (parallel count) in this flow"""
+        """Get the number of peas (parallel count) in this Flow
```
Comment (alexcg1) [inline jina/flow/base.py:485]: Peas

File: jina/flow/base.py [hunk 8]
```diff
@@ -449,20 +476,23 @@ def start(self):
...
+        # noqa: DAR201"""
         return sum(v.num_peas for v in self._pod_nodes.values())
 
-    def __eq__(self, other: 'BaseFlow'):
+    def __eq__(self, other: 'BaseFlow') -> bool:
         """
-        Comparing the topology of a flow with another flow.
+        Comparing the topology of a Flow with another Flow.
```
Comment (alexcg1) [inline jina/flow/base.py:491]: Compare

File: jina/flow/base.py [hunk 9]
```diff
@@ -618,10 +648,12 @@ def _ipython_display_(self):
         """Displays the object in IPython as a side effect"""
         self.plot(inline_display=True, build=(self._build_level != FlowBuildLevel.GRAPH))
 
-    def _mermaid_to_url(self, mermaid_str, img_type) -> str:
+    def _mermaid_to_url(self, mermaid_str: str, img_type: str) -> str:
         """
-        Rendering the current flow as a url points to a SVG, it needs internet connection
-        :param kwargs: keyword arguments of :py:meth:`to_mermaid`
+        Rendering the current Flow as a url points to a SVG, it needs internet connection
```
Comment (alexcg1) [inline jina/flow/base.py:653]: Render
URL

File: jina/flow/mixin/crud.py [hunk 10]
```diff
@@ -333,13 +345,12 @@ def update(self, input_fn: InputFnType,
                on_error: CallbackFnType = None,
                on_always: CallbackFnType = None,
                **kwargs):
-        """Updates documents on the current flow
+        """Updates documents on the current Flow
```
Comment (alexcg1) [inline jina/flow/mixin/crud.py:348]: Documents

**Review by alexcg1 — CHANGES_REQUESTED** [review]

**Comment by cristianmtr** [comment]
> Looks fine generally. Just check capitalization of Pods, Flows, Peas throughout (I've commented initially but don't have time to go through each one - there are a few tho)

Check again

**Pull request merged**